django_backend/scrounger/views.py

```diff
@@ -188,6 +191,21 @@ def search_view(request):
         sim_query = boonsdk.SimilarityQuery(simhashes)
         must_queries.append(sim_query)
 
+    # Check for a query param containing uploaded assets for similarity search.
+    uploaded_assets_json = request.GET.get('uploaded_assets')
+    if uploaded_assets_json:
+        images = []
+        uploaded_assets = json.loads(unquote(uploaded_assets_json))
+        for uploaded_asset in uploaded_assets:
+            image_format, image_base64 = uploaded_asset.split(';base64,')
+            image_decoded = base64.b64decode(image_base64)
+            image_file = NamedTemporaryFile(suffix='jpeg', delete=False)
+            image_file.write(image_decoded)
+            image_file.flush()
+            images.append(image_file.name)
```

Comment:
```suggestion
            filepath, headers = urlibretrive(uploaded_asset)
            images.append(filepath)
```

You might be able to get this whole block to 2 lines. import path is `from urllib.request import urlretrieve`

Reply:
it works indeed!